pkg/frontend/transport/handler.go

```diff
@@ -188,6 +190,18 @@ func (f *Handler) reportSlowQuery(r *http.Request, queryString url.Values, query
 	level.Info(util_log.WithContext(r.Context(), f.log)).Log(logMessage...)
 }
 
+func (f *Handler) reportFailedQueryStats(r *http.Request, queryString url.Values, queryResponseTime time.Duration, stats *querier_stats.Stats) {
+	// Log failed query info
+	logMessage := append([]interface{}{
+		"msg", "failed query stats",
+		"component", "query-frontend",
+		"response_time", queryResponseTime,
+		"sharded_queries", stats.LoadShardedQueries(),
```

Comment:
I'm wondering if the log fields here should be the same as for `reportSlowQuery`, except for `msg` and that we might want to add an `err` field? Is there a reason for having different log fields?

Reply:
If they're available, I agree.

Reply:
I was talking to @stevesg and these were the ones we thought would be most important right now. I can add the other ones in now if they will be useful

Reply:
I see now that these fields are copied from `reportQueryStats`, while I was comparing with `reportSlowQuery`. Maybe it would be better to add the "component" field to the logger, so it won't have to be included explicitly with every log entry?

Reply:
Sure thing, I will update.

Reply:
@pracucci did you mean to comment [here](https://github.com/grafana/mimir/pull/3190#discussion_r992694405)? :)

Reply:
Yes, sorry. I will stop commenting and I will grab a coffee :)